ImputationPipeline/Imputation.wdl

```diff
@@ -0,0 +1,475 @@
+          panel_bcf_index = path_to_reference_panel + "ALL.chr" + chrom + ".phase3_integrated.20130502.genotypes.bcf.csi",
+          sarah_jar = new_gatk_jar
+      }
+
+      call PrePhaseVariantsEagle {
+        input:
+          dataset_bcf = QConChunk.valid_chunk_bcf,
+          dataset_bcf_index = QConChunk.valid_chunk_bcf_index,
+          reference_panel_bcf = path_to_reference_panel + "ALL.chr" + chrom + ".phase3_integrated.20130502.genotypes.bcf",
+          reference_panel_bcf_index = path_to_reference_panel + "ALL.chr" + chrom + ".phase3_integrated.20130502.genotypes.bcf.csi",
+          chrom = chrom,
+          valid_chunk = QConChunk.valid,
+          genetic_map_file = genetic_maps_eagle,
+          start = (i * chunkLength) + 1,
+          end = (i + 1) * chunkLength + 5000000 # they do an overlap of 5,000,000 bases in michigan pipeline
+      }
+
+        call minimac4 {
+          input:
+            ref_panel = path_to_m3vcf + chrom + ".1000g.Phase3.v5.With.Parameter.Estimates.m3vcf.gz",
+            phased_vcf = PrePhaseVariantsEagle.dataset_prephased_vcf,
+            prefix = "chrom" + "_chunk_" + i +"_imputed",
+            chrom = chrom,
+            valid_phased_vcf = QConChunk.valid,
+            start = (i * chunkLength) + 1,
+            end = (i + 1) * chunkLength
+        }
+     }
+  }
+
+  Array[File] phased_vcfs = flatten(minimac4.vcf)
+
+  call MergeVCFs {
+    input:
+      input_vcfs = phased_vcfs,
+      output_vcf_basename = output_callset_name,
+      shard_size = size(minimac4.vcf[0], "GB")
+  }
+
+  File michiganserver_vcf = MergeVCFs.output_vcf
+
+  call UpdateHeader {
+    input:
+      vcf = michiganserver_vcf,
+      ref_dict = ref_dict
+  }
+
+  call RemoveSymbolicAlleles as RSAlleles {
+    input:
+      original_vcf = UpdateHeader.output_vcf,
+      original_vcf_index = UpdateHeader.output_vcf_index
+  }
+
+  call SeparateMultiallelics {
+    input:
+      original_vcf = RSAlleles.output_vcf,
+      original_vcf_index = RSAlleles.output_vcf_index
+  }
+
+  scatter (i in range(length(samples))) {
+
+    call SplitSample {
+      input: 
+        sample = samples[i],
+        vcf = SeparateMultiallelics.output_vcf,
+        vcf_index = SeparateMultiallelics.output_vcf_index
+    }
+  }
+  output {
+    Array[File] imputed_output_vcf = SplitSample.output_gzipped_vcf
+    Array[File] imputed_output_vcf_index = SplitSample.output_gzipped_vcf_index
+  }
+}
+
+task RemoveDuplicates {
+    input {
+      File input_vcfs
+      File input_vcf_indices
+      String output_basename
+    }
+     Int disk_size = 200
+  command <<<
+    bcftools norm -d both ~{input_vcfs} | bgzip -c > ~{output_basename}.vcf.gz
+    bcftools index -t ~{output_basename}.vcf.gz
+  >>>
+  runtime {
+    docker: "farjoun/impute:0.0.3-1504715575"
+    memory: "3 GiB"
+    disks: "local-disk " + disk_size + " HDD"
+  }
+  output {
```

Comment:
are there any metrics from minimac or eagle, or the qc of the input arrays that might be good to have as outputs of the workflow?

Reply:
I think Michigan just outputs which variants passed their QC thresholds, so I guess we could add that
There's the "info" file from Minimac which we output, although don't save as a final output of the workflow, it tells you some info about each site being imputed I think? I'm not sure how helpful it is

Reply:
this reminds me that we need to decide what files to output